examples/digital_fingerprinting/production/morpheus/dfp_azure_modules_pipeline.py

```diff
@@ -0,0 +1,393 @@
+
+    config.ae.feature_columns = load_labels_file(get_package_relative_file("data/columns_ae_azure.txt"))
+    config.ae.userid_column_name = "username"
+    config.ae.timestamp_column_name = "timestamp"
+
+    # Specify the column names to ensure all data is uniform
+    source_column_info = [
+        DateTimeColumn(name=config.ae.timestamp_column_name, dtype=datetime, input_name="time"),
+        RenameColumn(name=config.ae.userid_column_name, dtype=str, input_name="properties.userPrincipalName"),
+        RenameColumn(name="appDisplayName", dtype=str, input_name="properties.appDisplayName"),
+        ColumnInfo(name="category", dtype=str),
+        RenameColumn(name="clientAppUsed", dtype=str, input_name="properties.clientAppUsed"),
+        RenameColumn(name="deviceDetailbrowser", dtype=str, input_name="properties.deviceDetail.browser"),
+        RenameColumn(name="deviceDetaildisplayName", dtype=str, input_name="properties.deviceDetail.displayName"),
+        RenameColumn(name="deviceDetailoperatingSystem",
+                     dtype=str,
+                     input_name="properties.deviceDetail.operatingSystem"),
+        StringCatColumn(name="location",
+                        dtype=str,
+                        input_columns=[
+                            "properties.location.city",
+                            "properties.location.countryOrRegion",
+                        ],
+                        sep=", "),
+        RenameColumn(name="statusfailureReason", dtype=str, input_name="properties.status.failureReason"),
+    ]
+
+    source_schema = DataFrameInputSchema(json_columns=["properties"], column_info=source_column_info)
+
+    # Preprocessing schema
+    preprocess_column_info = [
+        ColumnInfo(name=config.ae.timestamp_column_name, dtype=datetime),
+        ColumnInfo(name=config.ae.userid_column_name, dtype=str),
+        ColumnInfo(name="appDisplayName", dtype=str),
+        ColumnInfo(name="clientAppUsed", dtype=str),
+        ColumnInfo(name="deviceDetailbrowser", dtype=str),
+        ColumnInfo(name="deviceDetaildisplayName", dtype=str),
+        ColumnInfo(name="deviceDetailoperatingSystem", dtype=str),
+        ColumnInfo(name="statusfailureReason", dtype=str),
+
+        # Derived columns
+        IncrementColumn(name="logcount",
+                        dtype=int,
+                        input_name=config.ae.timestamp_column_name,
+                        groupby_column=config.ae.userid_column_name),
+        CustomColumn(name="locincrement",
+                     dtype=int,
+                     process_column_fn=partial(create_increment_col, column_name="location")),
+        CustomColumn(name="appincrement",
+                     dtype=int,
+                     process_column_fn=partial(create_increment_col, column_name="appDisplayName")),
+    ]
+
+    preprocess_schema = DataFrameInputSchema(column_info=preprocess_column_info, preserve_columns=["_batch_id"])
+
+    encoding = "latin1"
```

Comment:
Why are we encoding it like this?

Reply:
Adding bytes to the module configuration yields attribute value as None. Pickle expects the string to have encoding in order to load the string. Otherwise it's throwing `TypeError: string argument without an encoding`